src/psplit.jl

```diff
@@ -0,0 +1,430 @@
+    end
+    return new_affexpr, constant
+end
+
+function _build_partitioned_expression(
+    expr::T,
+    partition_variables::Vector{<:JuMP.AbstractVariableRef}
+) where {T <: JuMP.GenericQuadExpr}
+    new_quadexpr = zero(T)
+    constant = JuMP.constant(expr)
+    for var in partition_variables
+        for (pair, coeff) in expr.terms
+            if pair.a == var && pair.b == var
+                JuMP.add_to_expression!(new_quadexpr, coeff, var, var)
+            end
+        end
+    end
+    new_aff, _ = _build_partitioned_expression(expr.aff, partition_variables)
+    JuMP.add_to_expression!(new_quadexpr, new_aff)
+    return new_quadexpr, constant
+end
+
+function _build_partitioned_expression(
+    expr::T,
+    partition_variables::Vector{<:JuMP.AbstractVariableRef}
+) where {T <: JuMP.AbstractVariableRef}
+
+    if expr in partition_variables
+        return expr, zero(T)
+    else
+        return zero(T), zero(T)
+    end
+end
+
+function _build_partitioned_expression(
+    expr::T,
+    partition_variables::Vector{<:JuMP.AbstractVariableRef}
+) where {T <: Number}
+    return expr, zero(T)
+end
+
+################################################################################
+#                              BOUND AUXILIARY
+################################################################################
+function _bound_auxiliary(
+    model::JuMP.AbstractModel,
+    v::JuMP.AbstractVariableRef,
+    func::JuMP.GenericAffExpr,
+    method::PSplit
+)   
+    T = JuMP.value_type(typeof(model))
+    lower_bound = zero(T)
+    upper_bound = zero(T)
+    for (var, coeff) in func.terms
+        if var != v
+            JuMP.is_binary(var) && continue
+            var_lb, var_ub = variable_bound_info(var)
+            if coeff > 0.0
+                lower_bound += coeff * var_lb
+                upper_bound += coeff * var_ub
+            else
+                lower_bound += coeff * var_ub
+                upper_bound += coeff * var_lb
+            end
+        end
+    end
+    JuMP.set_lower_bound(v, lower_bound)
+    JuMP.set_upper_bound(v, upper_bound)
+end    
+
+function _bound_auxiliary(
+    model::JuMP.AbstractModel,
+    v::JuMP.AbstractVariableRef,
+    func::Number,
+    method::PSplit
+) 
+    #Do nothing?
+end
+
+function _bound_auxiliary(
+    model::JuMP.AbstractModel,
+    v::JuMP.AbstractVariableRef,
+    func::JuMP.GenericQuadExpr,
+    method::PSplit
+) 
+    T = JuMP.value_type(typeof(model))
+    lower_bound = zero(T)
+    upper_bound = zero(T)
+    
+    # Handle linear terms
+    for (var, coeff) in func.aff.terms
+        if var != v
+            JuMP.is_binary(var) && continue
+            var_lb, var_ub = variable_bound_info(var)
+            if coeff > 0.0
+                lower_bound += coeff * var_lb
+                upper_bound += coeff * var_ub
+            else
+                lower_bound += coeff * var_ub
+                upper_bound += coeff * var_lb
+            end
+        end
+    end
```

Comment:
Is there a way to refactor so this isn't repeated code from the affine function above?